uboot-rockchip: fix RockPro64 boot from eMMC

mj22226 opened the pull request:
The RockPro64 has an issue with uboot v2021.01, it hangs on booting when an eMMC is used, SD is unaffected.
```
U-Boot TPL 2021.01 (Jan 17 2021 - 13:31:56)
Channel 0: LPDDR4, 50MHz
BW=32 Col=10 Bk=8 CS0 Row=16/15 CS=1 Die BW=16 Size=2048MB
Channel 1: LPDDR4, 50MHz
BW=32 Col=10 Bk=8 CS0 Row=16/15 CS=1 Die BW=16 Size=2048MB
256B stride
lpddr4_set_rate: change freq to 400000000 mhz 0, 1
lpddr4_set_rate: change freq to 800000000 mhz 1, 0
Trying to boot from BOOTROM
Returning to boot ROM...

U-Boot SPL 2021.01 (Jan 17 2021 - 13:31:56 +0000)
Trying to boot from MMC2


U-Boot 2021.01 (Jan 17 2021 - 13:31:56 +0000) OpenWrt

SoC: Rockchip rk3399
Reset cause: POR
Model: Pine64 RockPro64 v2.1
DRAM:  3.9 GiB
PMIC:  RK808 
MMC:   mmc@fe310000: 2, mmc@fe320000: 1, sdhci@fe330000: 0
Loading Environment from SPIFlash... SF: Detected gd25q128 with page size 256 Bytes, erase size 4 KiB, total 16 MiB
*** Warning - bad CRC, using default environment

In:    serial
Out:   serial
Err:   serial
Model: Pine64 RockPro64 v2.1
Net:   eth0: ethernet@fe300000
starting USB...
Bus usb@fe380000: USB EHCI 1.00
Bus usb@fe3a0000: USB OHCI 1.0
Bus usb@fe3c0000: USB EHCI 1.00
Bus usb@fe3e0000: USB OHCI 1.0
Bus dwc3: usb maximum-speed not found
Register 2000140 NbrPorts 2
Starting the controller
USB XHCI 1.10
scanning bus usb@fe380000 for devices... 1 USB Device(s) found
scanning bus usb@fe3a0000 for devices... 1 USB Device(s) found
scanning bus usb@fe3c0000 for devices... 1 USB Device(s) found
scanning bus usb@fe3e0000 for devices... 1 USB Device(s) found
scanning bus dwc3 for devices... 1 USB Device(s) found
       scanning usb for storage devices... 0 Storage Device(s) found
Hit any key to stop autoboot:  0 
switch to partitions #0, OK
mmc0(part 0) is current device
Scanning mmc 0:1...
Found U-Boot script /boot.scr
365 bytes read in 8 ms (43.9 KiB/s)
## Executing script at 00500000
54757 bytes read in 17 ms (3.1 MiB/s)
12638216 bytes read in 1212 ms (9.9 MiB/s)
## Flattened Device Tree blob at 01f00000
     Booting using the fdt blob at 0x1f00000
```
It seems that `CONFIG_USE_PREBOOT=y` is causing the boot problem with OpenWrt, disabling it restore the normal boot.

```
U-Boot TPL 2021.01 (Jan 17 2021 - 20:22:39)
Channel 0: LPDDR4, 50MHz
BW=32 Col=10 Bk=8 CS0 Row=16/15 CS=1 Die BW=16 Size=2048MB
Channel 1: LPDDR4, 50MHz
BW=32 Col=10 Bk=8 CS0 Row=16/15 CS=1 Die BW=16 Size=2048MB
256B stride
lpddr4_set_rate: change freq to 400000000 mhz 0, 1
lpddr4_set_rate: change freq to 800000000 mhz 1, 0
Trying to boot from BOOTROM
Returning to boot ROM...

U-Boot SPL 2021.01 (Jan 17 2021 - 20:22:39 +0000)
Trying to boot from MMC2


U-Boot 2021.01 (Jan 17 2021 - 20:22:39 +0000) OpenWrt

SoC: Rockchip rk3399
Reset cause: POR
Model: Pine64 RockPro64 v2.1
DRAM:  3.9 GiB
PMIC:  RK808 
MMC:   mmc@fe310000: 2, mmc@fe320000: 1, sdhci@fe330000: 0
Loading Environment from SPIFlash... SF: Detected gd25q128 with page size 256 Bytes, erase size 4 KiB, total 16 MiB
*** Warning - bad CRC, using default environment

In:    serial
Out:   serial
Err:   serial
Model: Pine64 RockPro64 v2.1
Net:   eth0: ethernet@fe300000
Hit any key to stop autoboot:  0 
switch to partitions #0, OK
mmc0(part 0) is current device
Scanning mmc 0:1...
Found U-Boot script /boot.scr
365 bytes read in 8 ms (43.9 KiB/s)
## Executing script at 00500000
54757 bytes read in 17 ms (3.1 MiB/s)
12638216 bytes read in 1211 ms (10 MiB/s)
## Flattened Device Tree blob at 01f00000
   Booting using the fdt blob at 0x1f00000
   Loading Device Tree to 00000000f1f16000, end 00000000f1f265e4 ... OK

Starting kernel ...

[    0.000000] Booting Linux on physical CPU 0x0000000000 [0x410fd034]
[    0.000000] Linux version 5.4.89 (builder@buildhost) (gcc version 8.4.0 (OpenWrt GCC 8.4.0 r15520-54bfebdca0)) #0 SMP PREEMPT Sun Jan 17 20:22:39 2021
[    0.000000] Machine model: Pine64 RockPro64
```

blocktrron reviewed (changes requested):
This rather looks like something triggers the device to enter a recovery mode (Fastboot / UMS), which is executed in the preboot stage. So this is not a fix but rather a workaround.

Can you try to find out what causes this by increasing the verbosity of the output?

gechu commented:
I first tried booting using eMMC. It failed as expected. Then I tried booting sdcard, and to my surprise it failed as well. The eMMC was not insetered.

I can help out testing. Needs some info how to increase log level though.

mj22226 commented:
> I can help out testing. Needs some info how to increase log level though.

@gechu did you build the firmware applying the fix?

mj22226 commented:
> Can you try to find out what causes this by increasing the verbosity of the output?

I increased the log level

```
U-Boot TPL 2021.01 (Jan 17 2021 - 23:49:14)
Channel 0: LPDDR4, 50MHz
BW=32 Col=10 Bk=8 CS0 Row=16/15 CS=1 Die BW=16 Size=2048MB
Channel 1: LPDDR4, 50MHz
BW=32 Col=10 Bk=8 CS0 Row=16/15 CS=1 Die BW=16 Size=2048MB
256B stride
lpddr4_set_rate: change freq to 400000000 mhz 0, 1
lpddr4_set_rate: change freq to 800000000 mhz 1, 0
Trying to boot from BOOTROM
Returning to boot ROM...
rk3399_syscon syscon@ff770000: set_state_simple op missing
rk3399_syscon syscon@ff320000: set_state_simple op missing
rockchip_rk3399_pinctrl pinctrl: set_state_simple op missing
rk3399_syscon syscon@ff330000: set_state_simple op missing
rockchip_rk3399_dmc dmc: set_state_simple op missing
rockchip_rk3399_cru clock-controller@ff760000: set_state_simple op missing
ns16550_serial serial@ff1a0000: set_state_simple op missing

U-Boot SPL 2021.01 (Jan 17 2021 - 23:49:14 +0000)
rockchip_rk3399_sdhci_5_1 sdhci@fe330000: set_state_simple op missing
rockchip_rk3288_spi spi@ff1d0000: set_state_simple op missing
jedec_spi_nor flash@0: set_state_simple op missing
rockchip_rk3288_dw_mshc mmc@fe320000: set_state_simple op missing
Trying to boot from MMC2


U-Boot 2021.01 (Jan 17 2021 - 23:49:14 +0000) OpenWrt

SoC: Rockchip rk3399
Reset cause: POR
Model: Pine64 RockPro64 v2.1
DRAM:  rockchip_rk3399_dmc dmc: set_state_simple op missing
3.9 GiB
rockchip_usb2phy usb2-phy@e450: subnode host-port
rockchip_usb2phy usb2-phy@e450: subnode otg-port
rockchip_usb2phy usb2-phy@e460: subnode host-port
rockchip_usb2phy usb2-phy@e460: subnode otg-port
rockchip_typec_phy phy@ff7c0000: subnode dp-port
rockchip_typec_phy phy@ff7c0000: Warning: dp-port not supported yet!
rockchip_typec_phy phy@ff7c0000: subnode usb3-port
rockchip_typec_phy phy@ff800000: subnode dp-port
rockchip_typec_phy phy@ff800000: Warning: dp-port not supported yet!
rockchip_typec_phy phy@ff800000: subnode usb3-port
rk3399_syscon syscon@ff770000: set_state_simple op missing
rk3399_syscon syscon@ff320000: set_state_simple op missing
rockchip_rk3399_pinctrl pinctrl: set_state_simple op missing
rockchip_rk3399_pmucru pmu-clock-controller@ff750000: set_state_simple op missing
pinconfig i2c0: set_state_simple op missing
pinconfig i2c0-xfer: set_state_simple op missing
pinconfig pmic: set_state_simple op missing
pinconfig pmic-int-l: set_state_simple op missing
PMIC:  RK808 
rk8xx_buck DCDC_REG1: set_state_simple op missing
rk8xx_buck DCDC_REG2: set_state_simple op missing
rk8xx_buck DCDC_REG3: set_state_simple op missing
rk8xx_buck DCDC_REG4: set_state_simple op missing
rk8xx_ldo LDO_REG1: set_state_simple op missing
rk8xx_ldo LDO_REG2: set_state_simple op missing
rk8xx_ldo LDO_REG3: set_state_simple op missing
rk8xx_ldo LDO_REG4: set_state_simple op missing
rk8xx_ldo LDO_REG5: set_state_simple op missing
rk8xx_ldo LDO_REG6: set_state_simple op missing
rk8xx_ldo LDO_REG7: set_state_simple op missing
rk8xx_ldo LDO_REG8: set_state_simple op missing
rk8xx_switch SWITCH_REG1: set_state_simple op missing
rk8xx_switch SWITCH_REG2: set_state_simple op missing
regulator_fixed vcc12v-dcin: set_state_simple op missing
regulator_fixed vcc1v8-s3: set_state_simple op missing
rockchip_gpio_bank gpio1@ff730000: set_state_simple op missing
pinconfig pcie: set_state_simple op missing
pinconfig pcie-pwr-en: set_state_simple op missing
regulator_fixed vcc3v3-sys: set_state_simple op missing
rockchip_gpio_bank gpio4@ff790000: set_state_simple op missing
pinconfig usb2: set_state_simple op missing
pinconfig vcc5v0-host-en: set_state_simple op missing
pinconfig usb-typec: set_state_simple op missing
pinconfig vcc5v0_typec_en: set_state_simple op missing
regulator_fixed vcc5v0-sys: set_state_simple op missing
regulator_fixed vcc5v0-usb: set_state_simple op missing
pinconfig pwm2: set_state_simple op missing
pinconfig pwm2-pin: set_state_simple op missing
pwm_regulator vdd-log: set_state_simple op missing
rockchip_rk3399_cru clock-controller@ff760000: set_state_simple op missing
pinconfig uart2c: set_state_simple op missing
pinconfig uart2c-xfer: set_state_simple op missing
MMC:   rockchip_rk3399_sdhci_5_1 sdhci@fe330000: set_state_simple op missing
pinconfig sdmmc: set_state_simple op missing
pinconfig sdmmc-clk: set_state_simple op missing
pinconfig sdmmc-cmd: set_state_simple op missing
pinconfig sdmmc-bus4: set_state_simple op missing
pinconfig sdio0: set_state_simple op missing
pinconfig sdio0-bus4: set_state_simple op missing
pinconfig sdio0-cmd: set_state_simple op missing
pinconfig sdio0-clk: set_state_simple op missing
mmc@fe310000: 2, mmc@fe320000: 1, sdhci@fe330000: 0
Loading Environment from SPIFlash... pinconfig spi1: set_state_simple op missing
pinconfig spi1-clk: set_state_simple op missing
pinconfig spi1-tx: set_state_simple op missing
pinconfig spi1-rx: set_state_simple op missing
pinconfig spi1-cs0: set_state_simple op missing
rockchip_rk3288_spi spi@ff1d0000: spi_find_chip_select: plat=00000000f1f3ae30, cs=0
jedec_spi_nor flash@0: set_state_simple op missing
SF: Detected gd25q128 with page size 256 Bytes, erase size 4 KiB, total 16 MiB
jedec_spi_nor flash@0: from 0x003f8000, len 32768
*** Warning - bad CRC, using default environment

rk3399_vop vop@ff8f0000: set_state_simple op missing
pinconfig i2c3: set_state_simple op missing
pinconfig i2c3-xfer: set_state_simple op missing
pinconfig hdmi: set_state_simple op missing
pinconfig hdmi-cec: set_state_simple op missing
In:    serial
Out:   vidconsole
Err:   vidconsole
Model: Pine64 RockPro64 v2.1
rockchip_efuse efuse@ff690000: set_state_simple op missing
rockchip_saradc saradc@ff100000: set_state_simple op missing
Net:   rockchip_gpio_bank gpio3@ff788000: set_state_simple op missing
pinconfig gmac: set_state_simple op missing
pinconfig rgmii-pins: set_state_simple op missing
fixed_clock external-gmac-clock: set_state_simple op missing
rockchip_reset reset: set_state_simple op missing
eth0: ethernet@fe300000
starting USB...
Bus usb@fe380000: ehci_generic usb@fe380000: set_state_simple op missing
ehci_generic usb@fe380000: No vbus supply
rockchip_usb2phy usb2-phy@e450: set_state_simple op missing
rockchip_usb2phy_port host-port: set_state_simple op missing
USB EHCI 1.00
Bus usb@fe3a0000: ohci_generic usb@fe3a0000: set_state_simple op missing
USB OHCI 1.0
Bus usb@fe3c0000: ehci_generic usb@fe3c0000: set_state_simple op missing
ehci_generic usb@fe3c0000: No vbus supply
rockchip_usb2phy usb2-phy@e460: set_state_simple op missing
rockchip_usb2phy_port host-port: set_state_simple op missing
USB EHCI 1.00
Bus usb@fe3e0000: ohci_generic usb@fe3e0000: set_state_simple op missing
USB OHCI 1.0
Bus dwc3: usb maximum-speed not found
No USB maximum speed specified. Using super speed
dwc3-generic-wrapper usb@fe900000: set_state_simple op missing
dwc3-generic-host dwc3: set_state_simple op missing
rockchip_usb2phy_port otg-port: set_state_simple op missing
rockchip_typec_phy phy@ff800000: set_state_simple op missing
rockchip_tcphy_usb3_port usb3-port: set_state_simple op missing
dwc3-generic-host dwc3: found 7 IN and 6 OUT endpoints
dwc3-generic-host dwc3: Event buf 00000000f1f4b0c0 dma f1f4b0c0 length 256
Register 2000140 NbrPorts 2
Starting the controller
USB XHCI 1.10
scanning bus usb@fe380000 for devices... usb_hub usb_hub: set_state_simple op missing
1 USB Device(s) found
scanning bus usb@fe3a0000 for devices... usb_hub usb_hub: set_state_simple op missing
1 USB Device(s) found
scanning bus usb@fe3c0000 for devices... usb_hub usb_hub: set_state_simple op missing
1 USB Device(s) found
scanning bus usb@fe3e0000 for devices... usb_hub usb_hub: set_state_simple op missing
1 USB Device(s) found
scanning bus dwc3 for devices... usb_hub usb_hub: set_state_simple op missing
1 USB Device(s) found
       scanning usb for storage devices... 0 Storage Device(s) found
Hit any key to stop autoboot:  0 
switch to partitions #0, OK
mmc0(part 0) is current device
Scanning mmc 0:1...
Found U-Boot script /boot.scr
365 bytes read in 8 ms (43.9 KiB/s)
## Executing script at 00500000
54757 bytes read in 17 ms (3.1 MiB/s)
12638216 bytes read in 1212 ms (9.9 MiB/s)
## Flattened Device Tree blob at 01f00000
   Booting using the fdt blob at 0x1f00000
```

gechu commented:
> > I can help out testing. Needs some info how to increase log level though.
> 
> @gechu did you build the firmware applying the fix?

No, I flashed a pre-built snapshot. Noticed that it was a sysupgrade image, but I assumed it would do since no factory image was available, perhaps a faulty assumption.

mj22226 commented:
> No, I flashed a pre-built snapshot. Noticed that it was a sysupgrade image,

@gechu There is not a factory image on rockchip target the system upgrade image can be used in both cases.
Try building the firmware applying this fix or with this firmware with the fix applied on it 
[openwrt-rockchip-armv8-pine64_rockpro64-squashfs-sysupgrade.img.gz](https://github.com/openwrt/openwrt/files/5844832/openwrt-rockchip-armv8-pine64_rockpro64-squashfs-sysupgrade.img.gz)

gechu commented:
> > No, I flashed a pre-built snapshot. Noticed that it was a sysupgrade image,
> 
> @gechu There is not a factory image on rockchip target the system upgrade image can be used in both cases.
> Try building the firmware applying this fix or with this firmware with the fix applied on it
> [openwrt-rockchip-armv8-pine64_rockpro64-squashfs-sysupgrade.img.gz](https://github.com/openwrt/openwrt/files/5844832/openwrt-rockchip-armv8-pine64_rockpro64-squashfs-sysupgrade.img.gz)

Thanks! Quickly tried it. Video output was messy and the connected TV couldn't handle it. Apart from that it booted just fine, discovered my Intel I350-T4, and nothing critical in the logs.

Is this firmware while still fresh stable enough as a daily driver?

I can provide more logs, just share the relevant commands.

blocktrron commented:
@mj22226 Can you share which value "preboot" contains?

mj22226 commented:
> @mj22226 Can you share which value "preboot" contains?

@blocktrron it seems that has no values
```
> env print preboot
preboot=usb start
=> printenv             
altbootcmd=setenv boot_syslinux_conf extlinux/extlinux-rollback.conf;run distro_bootcmd
arch=arm
baudrate=1500000
board=rockpro64_rk3399
board_name=rockpro64_rk3399
boot_a_script=load ${devtype} ${devnum}:${distro_bootpart} ${scriptaddr} ${prefix}${script}; source ${scriptaddr}
boot_efi_binary=load ${devtype} ${devnum}:${distro_bootpart} ${kernel_addr_r} efi/boot/bootaa64.efi; if fdt addr ${fdt_addr_r}; then bootefi ${kernel_addr_r} ${fdt_addr_r};else bootefi ${kernel_addr_r} ${fdtcontroladdr};fi
boot_efi_bootmgr=if fdt addr ${fdt_addr_r}; then bootefi bootmgr ${fdt_addr_r};else bootefi bootmgr;fi
boot_extlinux=sysboot ${devtype} ${devnum}:${distro_bootpart} any ${scriptaddr} ${prefix}${boot_syslinux_conf}
boot_net_usb_start=usb start
boot_pci_enum=pci enum
boot_prefixes=/ /boot/
boot_script_dhcp=boot.scr.uimg
boot_scripts=boot.scr.uimg boot.scr
boot_syslinux_conf=extlinux/extlinux.conf
boot_targets=mmc0 mmc1 nvme0 usb0 pxe dhcp sf0 
bootcmd=run distro_bootcmd
bootcmd_dhcp=setenv devtype dhcp; run boot_net_usb_start; run boot_pci_enum; if dhcp ${scriptaddr} ${boot_script_dhcp}; then source ${scriptaddr}; fi;setenv efi_fdtfile ${fdtfile}; setenv efi_old_vci ${bootp_vci};setenv efi_old_arch ${bootp_arch};setenv bootp_vci PXEClient:Arch:00011:UNDI:003000;setenv bootp_arch 0xb;if dhcp ${kernel_addr_r}; then tftpboot ${fdt_addr_r} dtb/${efi_fdtfile};if fdt addr ${fdt_addr_r}; then bootefi ${kernel_addr_r} ${fdt_addr_r}; else bootefi ${kernel_addr_r} ${fdtcontroladdr};fi;fi;setenv bootp_vci ${efi_old_vci};setenv bootp_arch ${efi_old_arch};setenv efi_fdtfile;setenv efi_old_arch;setenv efi_old_vci;
bootcmd_mmc0=devnum=0; run mmc_boot
bootcmd_mmc1=devnum=1; run mmc_boot
bootcmd_nvme0=devnum=0; run nvme_boot
bootcmd_pxe=run boot_net_usb_start; run boot_pci_enum; dhcp; if pxe get; then pxe boot; fi
bootcmd_sf0=busnum=0; run sf_boot
bootcmd_usb0=devnum=0; run usb_boot
bootdelay=2
cpu=armv8
cpuid#=54564d3738392e303000000000150e06
distro_bootcmd=setenv nvme_need_init; for target in ${boot_targets}; do run bootcmd_${target}; done
efi_dtb_prefixes=/ /dtb/ /dtb/current/
ethaddr=b6:de:ae:39:14:6b
fdt_addr_r=0x01f00000
fdtcontroladdr=f1f29990
fdtfile=rockchip/rk3399-rockpro64.dtb
kernel_addr_r=0x02080000
kernel_comp_addr_r=0x08000000
kernel_comp_size=0x2000000
load_efi_dtb=load ${devtype} ${devnum}:${distro_bootpart} ${fdt_addr_r} ${prefix}${efi_fdtfile}
mmc_boot=if mmc dev ${devnum}; then devtype=mmc; run scan_dev_for_boot_part; fi
nvme_boot=run boot_pci_enum; run nvme_init; if nvme dev ${devnum}; then devtype=nvme; run scan_dev_for_boot_part; fi
nvme_init=if ${nvme_need_init}; then setenv nvme_need_init false; nvme scan; fi
partitions=uuid_disk=${uuid_gpt_disk};name=loader1,start=32K,size=4000K,uuid=${uuid_gpt_loader1};name=loader2,start=8MB,size=4MB,uuid=${uuid_gpt_loader2};name=trust,size=4M,uuid=${uuid_gpt_atf};name=boot,size=112M,bootable,uuid=${uuid_gpt_boot};name=rootfs,size=-,uuid=B921B045-1DF0-41C3-AF44-4C6F280D3FAE;
preboot=usb start
pxefile_addr_r=0x00600000
ramdisk_addr_r=0x06000000
scan_dev_for_boot=echo Scanning ${devtype} ${devnum}:${distro_bootpart}...; for prefix in ${boot_prefixes}; do run scan_dev_for_extlinux; run scan_dev_for_scripts; done;run scan_dev_for_efi;
scan_dev_for_boot_part=part list ${devtype} ${devnum} -bootable devplist; env exists devplist || setenv devplist 1; for distro_bootpart in ${devplist}; do if fstype ${devtype} ${devnum}:${distro_bootpart} bootfstype; then run scan_dev_for_boot; fi; done; setenv devplist
scan_dev_for_efi=setenv efi_fdtfile ${fdtfile}; for prefix in ${efi_dtb_prefixes}; do if test -e ${devtype} ${devnum}:${distro_bootpart} ${prefix}${efi_fdtfile}; then run load_efi_dtb; fi;done;run boot_efi_bootmgr;if test -e ${devtype} ${devnum}:${distro_bootpart} efi/boot/bootaa64.efi; then echo Found EFI removable media binary efi/boot/bootaa64.efi; run boot_efi_binary; echo EFI LOAD FAILED: continuing...; fi; setenv efi_fdtfile
scan_dev_for_extlinux=if test -e ${devtype} ${devnum}:${distro_bootpart} ${prefix}${boot_syslinux_conf}; then echo Found ${prefix}${boot_syslinux_conf}; run boot_extlinux; echo SCRIPT FAILED: continuing...; fi
scan_dev_for_scripts=for script in ${boot_scripts}; do if test -e ${devtype} ${devnum}:${distro_bootpart} ${prefix}${script}; then echo Found U-Boot script ${prefix}${script}; run boot_a_script; echo SCRIPT FAILED: continuing...; fi; done
scan_sf_for_scripts=${devtype} read ${scriptaddr} ${script_offset_f} ${script_size_f}; source ${scriptaddr}; echo SCRIPT FAILED: continuing...
script_offset_f=0xffe000
script_size_f=0x2000
scriptaddr=0x00500000
serial#=09f14230a20fcc5f
sf_boot=if sf probe ${busnum}; then devtype=sf; run scan_sf_for_scripts; fi
soc=rk3399
stderr=serial,vidconsole
stdin=serial,usbkbd
stdout=serial,vidconsole
usb_boot=usb start; if usb dev ${devnum}; then devtype=usb; run scan_dev_for_boot_part; fi
vendor=pine64

Environment size: 4877/32764 bytes
=> 
```

blocktrron commented:
How did you increase the log level? I'm asking as there is a lot still missing.

The device seems to hang when relocating the FDT. You can try to set `fdt_high` to `0`, so the bootloader does not relocate the FDT. This would help to pinpoint the issue, even though there's still a lot missing from your logs.

mj22226 commented:
> How did you increase the log level? I'm asking as there is a lot still missing.
> 
> The device seems to hang when relocating the FDT. You can try to set `fdt_high` to `0`, so the bootloader does not relocate the FDT. This would help to pinpoint the issue, even though there's still a lot missing from your logs.

I add  to the rockpro64-rk3399_defconfig  `CONFIG_LOG=y CONFIG_LOG_MAX_LEVEL=9
[rock64pro-bootlog.txt](https://github.com/openwrt/openwrt/files/5858053/rock64pro-bootlog.txt)

`

blocktrron commented:
It looks like it's stuck upon disabling interrupts when stopping the USB controller (which is started in the preboot phase)

I didn't do more code diving there. I'm fine with merging as is, I'll just put a short explanation next to the patch explaining the root cause so we might have a chance of fixing this / identifying the downstream fix in the future.

Does this sound good to you?

mj22226 commented:
> I didn't do more code diving there. I'm fine with merging as is, I'll just put a short explanation next to the patch explaining the root cause so we might have a chance of fixing this / identifying the downstream fix in the future.
> 
> Does this sound good to you?

Yes is fine with me, this is a temporary fix.

mj22226 committed:
uboot-rockchip: fix RockPro64 boot from eMMC

On commit https://github.com/u-boot/u-boot/commit/f81f9f0ebac596bae7f27db095f4f0272b606cc3
CONFIG_USE_PREBOOT was enabled on the the RockPro64, this is causing boot issues
when a eMMC is used, as a workaround will temporarily disable this option.

Signed-off-by: Marty Jones <mj8263788@gmail.com>

blocktrron commented:
Merged, thanks.

> Yes is fine with me, this is a temporary fix.

Looking forward to the final fix to the issue.
